lambda/translcore.ml

```diff
@@ -1152,6 +1152,21 @@ and transl_exp0 ~in_new_scope ~scopes sort e =
           (* Might be a module, which are all values.  Otherwise raise. *)
           ignore (Env.find_module_lazy path e.exp_env)
       ) arg_idents;
+      let params, ap_args =
+        match param_idents with
+        | [] ->
+            [{name = Ident.create_local "unit";
+              layout = layout_probe_arg;
+              attributes = Lambda.default_param_attribute;
+              mode = alloc_local}]
```

Comment:
It'd be nice to factor this part out of the match to avoid duplication:

```
let make_param name = { 
   name; 
   layout = layout_probe_arg;               
   attributes = Lambda.default_param_attribute;
   mode = alloc_local }
in
...
```